### ISSUE TITLE
Support the Entry API

### ISSUE THREAD

**Issue by lfairy** [michaelsproul/rust_radix_trie#32]
See the [`HashMap` docs](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry).

**Comment by michaelsproul** [comment]
Good idea! This used to be in the `TODO` in the readme, but went missing during a move to issues...

Do you fancy having a crack at implementing it? There've been a lot of changes recently so most of my effort is directed towards testing and documenting in preparation for a release.
